### ISSUE TITLE
I do not want my code to be uploaded to the cloud!

### ISSUE THREAD

**Issue by stroobandt** [m5stack/micropython-m5stack#7]
I do not want my code to be uploaded to the cloud!!!
I want to keep certain code private.

**I am extremely uncomfortable** with sending my code or WiFi password to some cloud server.
Certain details of my code projects are strictly private.

Is there a way to still use M5Stack's MicroPython firmware/libraries whilst circumventing the cloud?
I prefer to program over a USB cable.

Please, provide firmware to allow the transfer of MicroPython programs over an USB cable without the need to connect to any WLAN.

**Comment by 0x1abin** [comment]
Hi, if you just want programs over an USB cable, you can remove `import wifisetup` and `import 
 m5cloud` at boot.py file.

**Comment by stroobandt** [comment]
Thanks.
However, I do not consider this to be a smart strategy for a company that wants to position itself as a prototyping platform for other companies.

You should at least put clear warnings up there, informing your users that their code will be uploaded to a cloud server.

**Comment by SuperMarioSF** [comment]
Hmmm...
If you doesn't use Web IDE, the file will not be uploaded to anywhere. File synchronize is only start on your operation in Web IDE.

**Comment by stroobandt** [comment]
That might be, but users should be informed and give their consent!

Anyhow, to overcome this issue, I flashed my M5Stacks with [LoBo MicroPython](https://github.com/loboris/MicroPython_ESP32_psRAM_LoBo/wiki) and installed Mika Tuupola's [M5Stack MicroPython Kitchen Sink](https://github.com/tuupola/micropython-m5stack).

**Comment by wegunterjr** [comment]
yes - this is old, but I agree - I plan to use this in offline mode, and some places may not have network access.  That is why M5G0 was good.  Can we have an offline use as well?
